tests/go/GofmtBearTest.py

```diff
@@ -1,7 +1,69 @@
+from queue import Queue
+
 from bears.go.GofmtBear import GofmtBear
-from coalib.testing.LocalBearTestHelper import verify_local_bear
+from coalib.testing.LocalBearTestHelper import LocalBearTestHelper, \
+    execute_bear
+
+from coalib.settings.Section import Section
+from coala_utils.ContextManagers import prepare_file
+
+
+in_file1 = """package main
+func main() {
+    return 1
+}"""
+
+out_file1 = """package main
+
+func main() {
+    return 1
+}
+"""
+
+simplify_in_file1 = """package main
+
+func main() {
+    array := []int{0, 2}
+    array2 := array[1:len(array)]
+    return 1
+}
+"""
+
+simplify_out_file1 = """package main
+
+func main() {
+    array := []int{0, 2}
+    array2 := array[1:]
+    return 1
+}
+"""
+
+in_file1 = in_file1.replace(' ' * 4, '\t')
+out_file1 = out_file1.replace(' ' * 4, '\t')
+simplify_in_file1 = simplify_in_file1.replace(' ' * 4, '\t')
+simplify_out_file1 = simplify_out_file1.replace(' ' * 4, '\t')
+
+
+class GofmtBearTest(LocalBearTestHelper):
+
+    def setUp(self):
+        self.section = Section('')
+        self.uut = GofmtBear(self.section, Queue())
+        self.maxDiff = None  # dbg
+
+    def test_without_simplify(self):
+        content = in_file1.splitlines()
+        with prepare_file(content, None) as (file, fname):
+            with execute_bear(self.uut, fname, file) as results:
+                fdict = {fname: file}
+                results[0].apply(fdict)
+                self.assertEqual(''.join(fdict[fname]), out_file1)
 
-GofmtBearTest = verify_local_bear(
-    GofmtBear,
-    ('package main\n\nfunc main() {\n\treturn 1\n}',),
-    ('package main\nfunc main() {\n\treturn 1\n}',))
+    def test_with_simplify(self):
+        content = simplify_in_file1.splitlines()
+        with prepare_file(content, None) as (file, fname):
+            with execute_bear(self.uut, fname, file, simplify='true'
+                              ) as results:
+                fdict = {fname: file}
+                results[0].apply(fdict)
+                self.assertEqual(''.join(fdict[fname]), simplify_out_file1)
```

Comment:
Can `self.check_results` be used here? What it does is that it runs the input with the bear and compares its `message` with the other. `Message` is not the same as output. I was using `self.check_results` here but that was not allowing me to check if input is being converted to the output or not. The only way to get output of a bear it seems is using the `apply` method which I have done here. Moreover this technique has been used in some other tests too.